File: LittleDemo/JetpackDemo/databinding/src/main/java/com/example/databinding/test/UserViewModel.java
package com.example.databinding.test;

import androidx.databinding.BaseObservable;
import androidx.databinding.Bindable;
import androidx.databinding.ObservableField;

import com.example.databinding.BR;
import com.example.databinding.entity.User;

public class UserViewModel  {

    private ObservableField<User> userObservableField;

    public UserViewModel(){
        User user = new User("Jack");
        userObservableField = new ObservableField<>();
        userObservableField.set(user);
    }

    public String getUserName(){
        return userObservableField.get().getUserName();
    }

    public void setUserName(String userName){
        userObservableField.get().setUserName(userName);
    }
}

File: LittleDemo/JetpackDemo/databinding/src/main/java/com/example/databinding/model/UserViewModel.kt
package com.example.databinding.model

import android.util.Log
import androidx.databinding.BaseObservable
import androidx.databinding.Bindable
import com.example.databinding.BR
import com.example.databinding.entity.User


class UserViewModel : BaseObservable(){


    var user: User = User("Jack")

    @Bindable
    fun getUserName(): String = user.userName

    fun setUserName(userName: String?){
        userName?.let {
            if (it != user.userName){
                user.userName = it
                Log.i("WLY","set username : $userName")
                notifyPropertyChanged(BR.userName)
            }

        }

    }
}

